test/v5/L2Deposit.ts

```diff
@@ -0,0 +1,357 @@
+                    sender: erc1271Wallet.address,
+                    recipient: erc1271Wallet.address,
+                    data: encodingHelper.encodeL2ArbitrumDepositData(
+                        wallet.user.address,
+                        arbitrumGasData,
+                    ),
+                }
+                // Obtain a specific amount of tokens and approve their to be used by the AllowanceTarget contract
+                await dealTokenAndApprove(
+                    wallet.user,
+                    tokenlon.AllowanceTarget,
+                    deposit.l1TokenAddr,
+                    defaultDeposit.amount,
+                    {
+                        walletContract: erc1271Wallet,
+                    },
+                )
+                // Add "\x19\x01" string to the data to be signed in order to avoid the risk of eth_sign.
+                const digest = await signingHelper.getL2DepositEIP712Digest(deposit, {
+                    chainId: network.chainId,
+                    verifyingContract: tokenlon.L2Deposit.address,
+                })
+                // Sign the digest data using the EOA
+                const digestSigned = await wallet.user.signMessage(ethers.utils.arrayify(digest))
+                // Convert the signed digest into the WalletBytes32 format.
+                const signature = signingHelper.composeSignature(
+                    digestSigned,
+                    SignatureType.WalletBytes32,
+                )
+                // Encode the deposit and signed data into a byte string with the function signatures hash prefix.
+                const payload = encodingHelper.encodeL2Deposit({
+                    deposit: deposit,
+                    depositSig: signature,
+                })
+                // Send a transaction to the L2Deposit contract through the UserProxy contract
+                const tx = await tokenlon.UserProxy.connect(wallet.user).toL2Deposit(payload, {
+                    value: calculateArbitrumCallValue(arbitrumGasData),
+                })
+                // Wait for the block to be mined and the transaction will be completed
+                const receipt = await tx.wait()
+                // Verify that the event outputs are equal to the deposit data parameters
+                assertEvent(receipt, deposit)
+            })
+        })
+
+        /* Test on the Optimism bridge */
+
+        describe("Deposit on the Optimism bridge", () => {
+            // Test for the EOA
+            it(`Should sign and encode a valid deposit for an EOA to use on the Optimism bridge.`, async () => {
+                const deposit = {
+                    // Override the default deposit data for the Optimism bridge
+                    ...defaultDeposit,
+                    l2Identifier: L2Identifier.Optimism,
+                    l1TokenAddr: token.USDT.address,
+                    l2TokenAddr: zeroAddress, // It does not matter for Optimism bridge
```

Comment:
Maybe we should use an actual address to prevent people reading the tests for reference thinking that they can just fill in zero address for Optimism deposit.